Fix updated analyzer issues and bump `snippets` version and changelog

TahaTesser opened the pull request:
fixes https://github.com/flutter/flutter/issues/127320

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

gspencergoog commented:
You'll have to swap instances of `Token.name` for `Token.lexeme` too, since the analyzer API changed.

TahaTesser commented on packages/snippets/pubspec.yaml:
```diff
@@ -2,10 +2,10 @@ name: snippets
+  sdk: '>=3.0.0 <4.0.0'
 
 dependencies:
   analyzer: 4.3.1
```
Thanks!

TahaTesser replied:
I think you meant the newer version of `file`  (not `file: ^6.0.0-nullsafety.4`) since the current version is already higher.

https://github.com/flutter/assets-for-api-docs/blob/0b2ee756f13c4be6923045f5dbf7cc27e973cd9d/packages/snippets/pubspec.yaml#L14

gspencergoog commented on packages/snippets/pubspec.yaml:
```diff
@@ -2,21 +2,21 @@ name: snippets
-  dart_style: ^2.2.3
-  file: ^6.1.0
+  dart_style: ^2.3.1
+  file: ^6.0.0-nullsafety.4
```
Actually, looks like I was wrong about this one: let's keep `^6.1.0`, which also appears to work.

TahaTesser replied:
`file: ^6.1.4` is the newer one, do we still want to keep `6.1.0`?

gspencergoog replied:
If `^6.1.4` works, then sure.  The latest version of `file` is `7.0.0`, which won't work.

TahaTesser replied:
`^6.1.4` seems to fail so I just reverted to `6.1.0` fingers crossed :)

gspencergoog reviewed (approved):
![32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d](https://user-images.githubusercontent.com/8867023/37737237-66e9c9dc-2d10-11e8-8797-13a405eace49.png)

gspencergoog commented:
Yes, that's because the flutter_plugin_tools version is out of date, I think.

Modify the `.cirrus.yml` file to refer to version `0.13.4+3` of `flutter_plugin_tools` and see if that fixes the error.  It'll cause other errors because there will be analyzer warnings to fix because it hasn't been updated in a while, but I think it'll fix that error at least.

TahaTesser commented:
@gspencergoog 
This is interesting, there were 4 issues. I've fixed 2 of them.

### `./ci/tool_runner format --fail-on-change --clang-format=clang-format-12`
Execute  `flutter_plugin_tools format` and fixed ✅


### `dart pub global run flutter_plugin_tools license-check`
Added missing license ✅

-----------------------

### `./ci/tool_runner analyze --custom-analysis=web_benchmarks/testing/test_app,flutter_lints/example`   ⏳

This is failing due to :
```console
Running command: "flutter pub get" in /tmp/cirrus-ci-build/packages/snippets
Running "flutter pub get" in snippets...                        
The current Dart SDK version is 2.19.0-146.2.beta.
Because snippets requires SDK version >=3.0.0 <4.0.0, version solving failed.
```

### ` ./ci/tool_runner pubspec-check` ⏳

This is more tricky, it expects the `snippets` example to be in the repository with `main` which doesn't have here. (flutter/packages defaults to `main` this rules in this repo)


 **current example**
`repository: https://github.com/flutter/assets-for-api-docs/tree/master/packages/snippets`

```console
./ci/tool_runner pubspec-check
Running for all packages, since a file has changed that could affect the entire repository.
Running for animation_metadata...
Running for diagram_capture...
Running for diagram_capture/example...
Running for diagram_generator...
Running for diagram_viewer...
Running for diagrams...
Running for snippets...
  The "repository" link should use "main", not "master".
Running for snippets/example...
```

I can verify this locally by updating the repository link but of course, this branch actually doesn't exist in `assets-for-api-docs`.
`repository: https://github.com/flutter/assets-for-api-docs/tree/main/packages/snippets` 
```console
➜  assets-for-api-docs git:(bump_dart_version) ✗ ./ci/tool_runner pubspec-check                                                                  
Running for all packages that have diffs relative to "cd876a777c20858812ee54fae32bee558c2c61f9"

No changed packages.
Running for animation_metadata...
Running for diagram_capture...
Running for diagram_capture/example...
Running for diagram_generator...
Running for diagram_viewer...
Running for diagrams...
Running for snippets...
Running for snippets/example...


------------------------------------------------------------
Ran for 8 package(s)


No issues found!
```

gspencergoog commented:
Ahh, I didn't realize we had added that check for `main`.  The repo will need to be branched to `main` first before you can fix that. I can do that if you like.

TahaTesser commented:
> Ahh, I didn't realize we had added that check for `main`. The repo will need to be branched to `main` first before you can fix that. I can do that if you like.

Appreciate it!

TahaTesser commented:
I also added a bunch of TODOs for deprecated APIs in some packages which were detected when fixing analyzer issues.

goderbauer commented on packages/diagram_capture/lib/diagram_capture.dart:
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// TODO(TahaTesser): Update this package for deprecated SingletonFlutterWindow API
+// https://github.com/flutter/flutter/issues/127585 and remove this ignore.
+// ignore_for_file: deprecated_member_use
```
would be better to just ignore the lines where this is used. Otherwise this will hide other deprecations in the future as well.

TahaTesser replied:
Thank you! That makes sense

goderbauer commented on packages/diagram_capture/lib/diagram_capture.dart:
```diff
@@ -598,7 +602,12 @@ class DiagramController {
       case ui.ImageByteFormat.rawStraightRgba:
         return 'RAW STRAIGHT RGBA';
+      case ui.ImageByteFormat.rawExtendedRgba128:
+        // TODO(TahaTesser): Handle this case.
```
should probably assert in case somebody accidentally runs into this?

goderbauer commented on packages/diagram_generator/lib/main.dart:
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// TODO(TahaTesser): Update this package for deprecated SingletonFlutterWindow API
+// https://github.com/flutter/flutter/issues/127586 and remove this ignore.
+// ignore_for_file: deprecated_member_use
```
same, better to add the ignore directly to the usage line.

goderbauer commented on packages/diagrams/lib/src/animation_diagram.dart:
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// TODO(TahaTesser): Update this package for deprecated SingletonFlutterWindow API
+// https://github.com/flutter/flutter/issues/127588 and remove this ignore.
+// ignore_for_file: deprecated_member_use
```
same

goderbauer commented on packages/diagrams/lib/src/curve.dart:
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// TODO(TahaTesser): Update this package for deprecated SingletonFlutterWindow API
+// https://github.com/flutter/flutter/issues/127588 and remove this ignore.
+// ignore_for_file: deprecated_member_use
```
same

TahaTesser commented:
Closing in favor of https://github.com/flutter/assets-for-api-docs/pull/219
